### ISSUE TITLE
Using the library in runtime

### ISSUE THREAD

**Issue by luispuig** [bobthered/tailwindcss-palette-generator#8]
Hi!

First of all great job with the library.

I tried the library as described and it works like a charm. It generates the pallet on build time in the file tailwind.config.js.

Now, I want to use it for a dynamic casa. I am importing the library in an AstroJS project (uses node) and want to generate the pallet from a colour from the DB. When I try to import the library I get this error:

```
16:35:01 [ERROR] Failed to resolve entry for package "@bobthered/tailwindcss-palette-generator". The package may have incorrect main/module/exports specified in its package.json: No known conditions for "." specifier in "@bobthered/tailwindcss-palette-generator" package
  Stack trace:
    at packageEntryFailure (file:///Users/lpuig/Development/turisapps_templates/template-leon/node_modules/vite/dist/node/chunks/dep-CDnG8rE7.js:46575:15)
```

**Comment by bobthered** [comment]
I've never used Astro.  Do you have a repo available for me to look at?

**Comment by luispuig** [comment]
I do not have a public project, but it is related to any node project. I did this for you to check it: https://stackblitz.com/edit/node-qzw1yd?file=index.ts

This code is working:

<img width="1478" alt="image" src="https://github.com/user-attachments/assets/37a08475-6a9f-4c3f-b05f-d84ee7005f95">


But, when trying to import the library, you need to reload the page for the script to run again:

<img width="1482" alt="image" src="https://github.com/user-attachments/assets/7cc4bdda-7df8-4f34-b56c-eb85294985d3">

**Comment by bobthered** [comment]
Thanks.  My initial thoughts are you're importing it as a common JS import in the second image, but the rest of the file is ES6 imports ( "require" vs. "import" ).

Have you tried using the "import"?

**Comment by luispuig** [comment]
I also tried `const {
  tailwindcssPaletteGenerator,
} = require("@bobthered/tailwindcss-palette-generator");`  and `import { tailwindcssPaletteGenerator } from '@bobthered/tailwindcss-palette-generato`.


I think it is something related with CommonJS / ES modules thing, but I do not have much experience with the differences and the build system for libs.

**Comment by bobthered** [comment]
I updated the build step to have both `csj` & `esm` exports.  New version is `3.3.2`.  See stackblitz example below.  Let me know if that works.

[https://stackblitz.com/edit/node-z6g7hv?file=package.json](https://stackblitz.com/edit/node-z6g7hv?file=package.json)

![image](https://github.com/user-attachments/assets/40cc9a31-a2e9-4feb-9cd0-55e363869bbb)

**Comment by luispuig** [comment]
Thanks a lot. I can see it works in the stack blitz with node and express. Unfortunately, it is not working completely on my project. AstroJS runs on top of Vite, probably my fault thinking it would work as in the node.

From my side, I copied your code to my project and it works great. It is not too much code. I am not stacked.

Only in case you would like to provide more flexibility to the library, I created a repo with an empty AstroJS to help: https://github.com/luispuig/test-tailwindcss-palette-generator/blob/main/src/pages/palette.ts

The error is:

```
Named export 'tailwindcssPaletteGenerator' not found. The requested module '@bobthered/tailwindcss-palette-generator' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from '@bobthered/tailwindcss-palette-generator';
const {tailwindcssPaletteGenerator} = pkg;
```

By the way, the proposal on the error message works.

Let me know if I can provide some extra help. Thanks!
